File: retroFitToDo/app/src/main/java/com/example/retrofittodo/MainActivity.java
package com.example.retrofittodo;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.util.Log;
import android.view.View;

import java.util.List;

import retrofit2.Call;
import retrofit2.Callback;
import retrofit2.Response;

public class MainActivity extends AppCompatActivity {
    private static final String TAG = "MainActivity";

    ApiInterface apiInterface;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        apiInterface = ApiClient.getClient().create(ApiInterface.class);
    }

    public void getToDo(View view) {

        Call<List<ToDo>> call = apiInterface.getToDo();
        call.enqueue(new Callback<List<ToDo>>() {
            @Override
            public void onResponse(Call<List<ToDo>> call, Response<List<ToDo>> response) {
                Log.e(TAG, "onResponse: "+ response.body() );

            }

            @Override
            public void onFailure(Call<List<ToDo>> call, Throwable t) {
                Log.e(TAG, "onFailure: "+ t.getLocalizedMessage() );
            }
        });

    }

    public void getTodoRoute(View view) {
        Call<ToDo> toDoRoute = apiInterface.getToDoRoute(1);
            toDoRoute.enqueue(new Callback<ToDo>() {
                @Override
                public void onResponse(Call<ToDo> call, Response<ToDo> response) {
                    Log.e(TAG, "onResponse: "+ response.body());

                }

                @Override
                public void onFailure(Call<ToDo> call, Throwable t) {
                    Log.e(TAG, "onFailure: "+ t.getLocalizedMessage() );

                }
            });

    }

    public void getToDoQuery(View view) {
        Call<ToDo> todoQuery = apiInterface.getToDoQuery(2,false);
        todoQuery.enqueue(new Callback<ToDo>() {
            @Override
            public void onResponse(Call<ToDo> call, Response<ToDo> response) {
                Log.e(TAG, "onResponse: "+response.body() );
            }

            @Override
            public void onFailure(Call<ToDo> call, Throwable t) {
                Log.e(TAG, "onFailure: "+ t.getLocalizedMessage() );
            }
        });
    }

    public void PostToDo(View view) {
        ToDo toDo = new ToDo(3, "get me a milk" ,false);
        Call<ToDo> toDoPostCall = apiInterface.postTodo(toDo);
        toDoPostCall.enqueue(new Callback<ToDo>() {
            @Override
            public void onResponse(Call<ToDo> call, Response<ToDo> response) {
                Log.e(TAG, "onResponse: "+ response.body() );
            }

            @Override
            public void onFailure(Call<ToDo> call, Throwable t) {
                Log.e(TAG, "onFailure: "+t.getLocalizedMessage() );

            }
        });
    }
}

File: cyber_nerd/CypherX/app/src/main/java/com/smartherd/cyber_nerd/MainActivity.kt
package com.cybernerd.cypherx

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.util.Log
import com.cybernerd.cypherx.model.CypherXApi
import com.cybernerd.cypherx.model.Department
import kotlinx.android.synthetic.main.activity_main.*
import retrofit2.Call
import retrofit2.Callback
import retrofit2.Response
import retrofit2.Retrofit
import retrofit2.converter.gson.GsonConverterFactory
import java.lang.StringBuilder

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        // retrofit builder
        // objects to call

        val retrofit = Retrofit.Builder()
            .baseUrl("https://ashutoshwahane.github.io/")
            .addConverterFactory(GsonConverterFactory.create())
            .build()
        val cypherXApi = retrofit.create(CypherXApi::class.java)
        val department_call:Call<List<Department>> = cypherXApi.getDepartment()

        department_call.enqueue(object :Callback<List<Department>>{
            override fun onFailure(call: Call<List<Department>>, t: Throwable) {
                Log.e(" Something went wrong",t.message.toString())
            }

            override fun onResponse(
                call: Call<List<Department>>,
                response: Response<List<Department>>
            ) {
               val departments:List<Department> = response.body()!!
               val stringBuilder = StringBuilder()

                for(department in departments){
                    stringBuilder.append(department.department_id)
                    stringBuilder.append("\n")
                    stringBuilder.append(department.department_name)
                    stringBuilder.append("\n")
                    stringBuilder.append(department.description)
                    stringBuilder.append("\n")
                }

                textview_title.text = stringBuilder
            }
        })

    }
}

